main/internal/implementations/markdowncreation/resources_outside_terraform_control.go

```diff
@@ -8,14 +8,88 @@ import (
 )
 
 // setResourcesOutsideOfTerraformControlData sets the resources outside terraform control data in the markdown report
-func (m *MarkdownCreator) setResourcesOutsideOfTerraformControlData(report *doc.MarkDownDoc) {
+func (m *MarkdownCreator) setResourcesOutsideOfTerraformControlData(report *doc.MarkDownDoc, newResourcesCostEstimation string) {
 	report.Write("# Resources Outside of Terraform Control").Writeln().Writeln()
 
 	if len(m.newResources) == 0 {
 		report.Write("No new resources found!").Writeln()
 		return
 	}
 
+	if len(m.costEstimates) == 0 || newResourcesCostEstimation == "0.00" {
+		m.resourcesWithoutCostEstimates(report)
+		return
+	}
+
+	m.resourcesWithCostEstimates(report)
+}
+
+// ResourceCostEstimate represents the cost estimate for a resource
+type ResourceCostEstimate struct {
+	ResourceCount  int
+	CostComponents map[string]bool
+	IsUsageBased   bool
+	MonthlyCost    float64
+}
+
+// resourcesWithCostEstimates sets the resources outside terraform control data in the markdown report
+func (m *MarkdownCreator) resourcesWithCostEstimates(report *doc.MarkDownDoc) {
+	resourcesDetailsByType := make(map[string]*ResourceCostEstimate)
+	for resourceID := range m.newResources {
+		resourceType := strings.Split(resourceID, ".")[0]
+		if _, ok := resourcesDetailsByType[resourceType]; !ok {
+			resourcesDetailsByType[resourceType] = &ResourceCostEstimate{}
+		}
+
+		resourcesDetailsByType[resourceType].ResourceCount++
+	}
+
+	for _, costEstimate := range m.costEstimates {
+		resourceType := strings.Split(costEstimate.ResourceName, ".")[0]
+
+		if _, ok := resourcesDetailsByType[resourceType]; ok {
+			if resourcesDetailsByType[resourceType].CostComponents == nil {
+				resourcesDetailsByType[resourceType].CostComponents = make(map[string]bool)
+			}
+
+			resourcesDetailsByType[resourceType].CostComponents[costEstimate.CostComponent] = true
+			resourcesDetailsByType[resourceType].IsUsageBased = costEstimate.IsUsageBased
+
+			resourcesDetailsByType[resourceType].MonthlyCost += m.getActualMonthlyCostEstimate(costEstimate)
+		}
+	}
+
+	report.Write("|Type|# Resources|Cost Components|Monthly Cost|Usage Based*|").Writeln()
+	report.Write("| :---: | :---: | :---: | :---: | :---: |").Writeln()
+
+	rowCounter := 0
```

Comment:
If not used within the for loop or to control its behavior we can delete this variable.